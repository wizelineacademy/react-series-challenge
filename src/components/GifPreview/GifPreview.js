import React from 'react';

import GitPreviewStyled, {GifPreviewDivContainter} from './GifPreviewStyled';
import FavoritePin from '../FavoritePin';
import {withRouter} from 'react-router';
import {DETAILS_PATH} from '../../config/routerConstants';

const GifPreview = ({gifDetails, alt, marked, history, escaleFactor = 1}) => {
  return (
    <GifPreviewDivContainter onClick={() => history.push(`${DETAILS_PATH}/${gifDetails.id}`)}>
      <FavoritePin gifReference={gifDetails} marked={marked}/>
      <GitPreviewStyled src={gifDetails.images.preview_gif.url}
        alt={alt}
        width={gifDetails.images.preview_gif.width * escaleFactor + 'px'}
        height={gifDetails.images.preview_gif.height * escaleFactor + 'px'}/>
    </GifPreviewDivContainter>
  );
};

export default withRouter(GifPreview);
