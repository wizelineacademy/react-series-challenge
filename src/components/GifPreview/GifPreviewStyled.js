import styled from 'styled-components';

const GifPreviewDivContainter = styled.div`
  position: relative;
`;

const GifPreviewStyled = styled.img`
  width: ${props => props.width ? props.width : ''};
  height: ${props => props.height ? props.height : ''};
`;

export {GifPreviewDivContainter};
export default GifPreviewStyled;
