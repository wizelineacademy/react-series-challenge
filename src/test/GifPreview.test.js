import React from 'react';
import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import {BrowserRouter as Router} from 'react-router-dom';

import GifPreview from '../components/GifPreview';

const mockedGif = {
  id: 'mockId',
  images: {
    preview_gif: {
      width: 200,
      height: 200
    }
  }
};

describe('GifPreview', () => {
  it('Should render correctly', () => {
    const output = shallow(
      <GifPreview/>
    );
    expect(shallowToJson(output)).toMatchSnapshot();
  });

  it ('Should render unmarked gif', () => {
    const output = shallow(
      <GifPreview gifDetails={mockedGif} alt="test" marked={false}/>
    );
    expect(shallowToJson(output)).toMatchSnapshot();
  })

  it ('Should render marked gif', () => {
    const output = shallow(
      <GifPreview gifDetails={mockedGif} alt="test" marked={true}/>
    );
    expect(shallowToJson(output)).toMatchSnapshot();
  });

  it ('Should render escaled gif x2', () => {
    const output = shallow(
      <GifPreview gifDetails={mockedGif} alt="test" marked={true}
        escaleFactor={2}/>
    );
    expect(shallowToJson(output)).toMatchSnapshot();
  });

  it('Should simulate a click', () => {
    const output = shallow(
      <GifPreview gifDetails={mockedGif} alt="test" marked={false}/>
    );
    output.simulate('click');
    expect(shallowToJson(output)).toMatchSnapshot();
  });

  it('Should simulate a click with router', () => {
    const output = shallow(
      <Router>
        <GifPreview gifDetails={mockedGif} alt="test" marked={false}/>
      </Router>
    );
    output.simulate('click');
    expect(shallowToJson(output)).toMatchSnapshot();
  });
});
