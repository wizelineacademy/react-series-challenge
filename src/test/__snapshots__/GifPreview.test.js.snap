// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GifPreview Should render correctly 1`] = `
<Route
  render={[Function]}
/>
`;

exports[`GifPreview Should render escaled gif x2 1`] = `
<Route
  render={[Function]}
/>
`;

exports[`GifPreview Should render marked gif 1`] = `
<Route
  render={[Function]}
/>
`;

exports[`GifPreview Should render unmarked gif 1`] = `
<Route
  render={[Function]}
/>
`;

exports[`GifPreview Should simulate a click 1`] = `
<Route
  render={[Function]}
/>
`;

exports[`GifPreview Should simulate a click with router 1`] = `
<Router
  history={
    Object {
      "action": "POP",
      "block": [Function],
      "createHref": [Function],
      "go": [Function],
      "goBack": [Function],
      "goForward": [Function],
      "length": 1,
      "listen": [Function],
      "location": Object {
        "hash": "",
        "pathname": "/",
        "search": "",
        "state": undefined,
      },
      "push": [Function],
      "replace": [Function],
    }
  }
>
  <withRouter(GifPreview)
    alt="test"
    gifDetails={
      Object {
        "id": "mockId",
        "images": Object {
          "preview_gif": Object {
            "height": 200,
            "width": 200,
          },
        },
      }
    }
    marked={false}
  />
</Router>
`;
